File: images/etl/postgres-to-elastic/src/db/elastic.py
import logging
from os import environ

from elasticsearch import Elasticsearch

logger = logging.getLogger('elastic')

elastic_host: str = environ.get('ELASTIC_HOST', 'elastic')
elastic_port: str = str(environ.get('ELASTIC_PORT', '9200'))
elastic_user: str = environ.get('ELASTIC_USER', 'elastic')
elastic_password: int = environ.get('ELASTIC_PASSWORD', '')

elastic = Elasticsearch(
    f'http://{elastic_host}:{elastic_port}',
    http_auth=(elastic_user, elastic_password),
)


def get_elastic() -> Elasticsearch:
    return elastic

File: images/async-api/pkg/database/elastic.go
package database

import (
	"async-api/internal/config"
	"fmt"
	"github.com/elastic/go-elasticsearch/v9"
	"log"
)

func SetupElasticClient(cfg config.Config) (*elasticsearch.Client, error) {
	esCfg := elasticsearch.Config{
		Addresses: []string{"http://" + cfg.ElasticHost + ":" + cfg.ElasticPort},
		Username:  cfg.ElasticUser,
		Password:  cfg.ElasticPass,
	}

	es, err := elasticsearch.NewClient(esCfg)
	if err != nil {
		return nil, fmt.Errorf("Error creating client: %s", err)
	}

	log.Println(es.Info())

	return es, nil
}
